app/models/concerns/host_info.rb

```diff
@@ -0,0 +1,24 @@
+module HostInfo
+  extend ActiveSupport::Concern
+
+  def info
+    info_hash = {}
+
+    self.class.info_providers.each do |provider_method|
```

Comment:
Also I might be missing something here but is this about the ENC or is this about something else?  `.info` has been used just for the ENC for a long time,  this seems to add more stuff to it, which is only going to be used by Puppet but makes it look like `.info` is used for other stuff. Is that the case?